### PR TITLE
Fix `minOccurrences` and `maxOccurrences` Handling in X12 to Ballerina Schema Conversion

### PR DESCRIPTION
**Pull request by RDPerera**
## Purpose
Fixes: https://github.com/ballerina-platform/ballerina-library/issues/6658 , https://github.com/ballerina-platform/ballerina-library/issues/6610

This PR addresses two issues in the EDI tool:

1. **Incorrect Handling of `minOccurrences` and `maxOccurrences` in Schema.json from X12 XSD Files:** 
This fix resolves an issue where maxOccurrences/minOccurrences values from X12 XSD files were not accurately converted into schema.json, leading all values to default to 1 irrespective of the defined occurrences in the XSD.

This is a part of the sample X12 file; 

**Before Fix:**  Values of maxOccurrences and minOccurrences are not properly converted into Ballerina EDI schema.

**After Fix:** The same segment now correctly converts to schema.json with maxOccurrences and minOccurrences.

2. **Enabling Documentation Comments in Generated Code:** 
Add documentation comments to the automatically generated public functions, enhancing the developer experience of the connector user.

**Before Fix:** 

**After Fix:**

### CODE REVIEW

File: edi-tools/modules/codegen/maincode_gen.bal [hunk 1]
```diff
@@ -37,6 +42,11 @@ public isolated function fromEdiString(string ediText, EDI_NAME ediName) returns
     return ediDeserialize(ediText);
 }
 
+# Convert Ballerina record to EDI string.
+# 
+# + data - Ballerina record to be converted
+# + ediName - EDI type name
+# + return - EDI string
```
Comment (aashikam) [inline edi-tools/modules/codegen/maincode_gen.bal:49]: Shouldn't this be string or error? it just says EDI string

File: edi-tools/modules/codegen/maincode_gen.bal [hunk 2]
```diff
@@ -29,6 +29,11 @@ public isolated function getEDINames() returns string[] {
     return ${libdata.ediNames.toString()};
 }
 
+# Convert EDI string to Ballerina record.
+# 
+# + ediText - EDI string to be converted
+# + ediName - EDI type name
+# + return - Ballerina record
```
Comment (aashikam) [inline edi-tools/modules/codegen/maincode_gen.bal:36]: I see the return type as anydata|error so this doc comment is confusing

File: edi-tools/modules/codegen/rest_connector_gen.bal [hunk 3]
```diff
@@ -31,7 +36,11 @@ service /${re.replaceAll(libName,"_")}Parser on new http:Listener(restConnectorP
         anydata target = check fromEdiString(ediData, ediTypeName);   
         return target.toJson();
     }
-
+    # Resource to convert Ballerina record to EDI string.
+    # 
+    # + ediType - EDI type name.
+    # + jsonData - Ballerina record as JSON.
+    # + return - EDI string.
```
Comment (aashikam) [inline edi-tools/modules/codegen/rest_connector_gen.bal:43]: Remove full stop at the end.

File: edi-tools/modules/codegen/schema_code_gen.bal [hunk 4]
```diff
@@ -28,26 +28,47 @@ public function generateCodeForSchema(json schema, string outputPath) returns er
...
+# + data - Ballerina record to be converted
+# + return - EDI string
 public isolated function toEdiString(${ediSchema.name} data) returns string|error {
     edi:EdiSchema ediSchema = check edi:getSchema(schemaJson);
     return edi:toEdiString(data, ediSchema);    
 } 
 
+# Get the EDI schema.
+# 
+# + return - EDI schema
```
Comment (aashikam) [inline edi-tools/modules/codegen/schema_code_gen.bal:52]: Same as other places the error is ignored in docs

File: edi-tools/modules/codegen/maincode_gen.bal (hunk 1 above)
Comment (RDPerera) [reply edi-tools/modules/codegen/maincode_gen.bal]: Changed with 1330f90e144a8514f5a75f7fd1d041a44035e7ba.

File: edi-tools/modules/codegen/maincode_gen.bal (hunk 2 above)
Comment (RDPerera) [reply edi-tools/modules/codegen/maincode_gen.bal]: Changed with 1330f90e144a8514f5a75f7fd1d041a44035e7ba.

File: edi-tools/modules/codegen/rest_connector_gen.bal (hunk 3 above)
Comment (RDPerera) [reply edi-tools/modules/codegen/rest_connector_gen.bal]: Changed with 1330f90e144a8514f5a75f7fd1d041a44035e7ba.

File: edi-tools/modules/codegen/schema_code_gen.bal (hunk 4 above)
Comment (RDPerera) [reply edi-tools/modules/codegen/schema_code_gen.bal]: Changed with 1330f90e144a8514f5a75f7fd1d041a44035e7ba.

**Comment by gayaldassanayake** [comment]
Tested for X12 810, 850 examples, and works as expected.

**Review by niveathika — APPROVED** [review]

**Pull request merged**